nevergrad/functions/corefuncs.py

```diff
@@ -6,50 +6,65 @@
 import typing as tp
 from math import exp, sqrt, tanh
 import numpy as np
-from nevergrad.parametrization import discretization
 from nevergrad.common.decorators import Registry
 
 
 registry: Registry[tp.Callable[[np.ndarray], float]] = Registry()
 
 
-def onemax(x: tp.List[int]) -> float:
-    """onemax(x) is the most classical case of discrete functions, adapted to minimization.
+class DiscreteFunction:
```

Comment:
If this is to be used independently of `ArtificialFunction` they why not make it an `ExperimentFunction` directly?

Reply:
I think this would not work.
An ExperimentFunction has a fixed dimension, whereas a DiscreteFunction is dimension-independent.

Reply:
It can work, the `ExperimentFunction` can take the dimension as input, just like the `ArtificialFunction` does.

`def __init__(name: str, dimension: int, arity: int)`

Then you have all you need. The dimension and arity will be used by the parametrization, and the name and arity by the underlying function. In `experiments.py`, that will take 1 line to define with 1 unique object instead of 3 with 3 different objects.